### PR TITLE
TBD-4738: Add db mapping for BigQuery spark components

### PR DESCRIPTION
**Pull request by bchen-talend**

### CODE REVIEW

File: main/plugins/org.talend.core.runtime/mappings/mapping_BigQuery.xml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0"?>
```
Comment (amarouni) [inline main/plugins/org.talend.core.runtime/mappings/mapping_BigQuery.xml]: Where is this file used ?

File: main/plugins/org.talend.core.runtime/mappings/mapping_BigQuery.xml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    <dbType type="INTEGER" default="true" />
+                </talendType>
+                <talendType type="id_Object">
+                </talendType>    
+                <talendType type="id_Short">
+                </talendType>
+                <talendType type="id_String">
+                    <dbType type="STRING" default="true" />
+                    <dbType type="DATETIME" />
+                    <dbType type="DATE" />
```
Comment (amarouni) [inline main/plugins/org.talend.core.runtime/mappings/mapping_BigQuery.xml]: Why is the `DATE` mapped to `String` ? Why cannot we map it to `Date` (`java.util.Date`) ?

File: main/plugins/org.talend.core.runtime/mappings/mapping_BigQuery.xml (hunk 1 above)
Comment (bchen-talend) [inline main/plugins/org.talend.core.runtime/mappings/mapping_BigQuery.xml]: For tBigQueryOutput component, which need to build bigquery table schema for insert & create table, also for writing jsonObject to insert.

File: main/plugins/org.talend.core.runtime/mappings/mapping_BigQuery.xml (hunk 2 above)
Comment (bchen-talend) [inline main/plugins/org.talend.core.runtime/mappings/mapping_BigQuery.xml]: As BigQuery's date type support subsecond, https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#datetime-type, so use String for the moment.

**Pull request merged**